File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (int t = 0; t < commands.size(); t++){
        int i = commands[t][0];
        int j = commands[t][1];
        int k = commands[t][2];
        
        vector<int> tmp(array.begin() + (i-1), array.begin() + j);
        sort(tmp.begin(), tmp.end());
        answer.push_back(tmp[k-1]);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] result = new int[commands.length];
        int idx = 0;
        for (int[] command : commands) {
            int i = command[0];
            int j = command[1];
            int k = command[2];
            
            int[] sub = Arrays.copyOfRange(array, i-1, j);
            Arrays.sort(sub);
            result[idx++] = sub[k-1];
        }
        return result;
    }
}